### ISSUE TITLE
IsKeyPressed different behaviour on the web than desktop after EndDrawing()

### ISSUE THREAD

**Issue by kallehed** [raysan5/raylib#3167]
### Issue description

When running Raylib in the browser,  there will be different behaviour with the IsKeyPressed function after EndDrawing(). On desktop, it will return that the some key is pressed when pressed, but in the browser, no keys will be pressed.

In the code below, "works on both" will be printed on both platforms, but "does not work in browser" will only be printed on desktop.

Basically, it seems like raylib clears data for IsKeyPressed() after EndDrawing() in the browser, while it doesn't on desktop for some reason. (I prefer not clearing)

One other interesting thing: IsKeyDown() has the same behaviour on both platforms (both logs are printed on both). So this must surely be a bug?

### Environment

Tested on Linux and Firefox. OpenGL 4.6, intel.

### Issue Screenshot
Firefox: 
![image](https://github.com/raysan5/raylib/assets/28019266/c907d825-74e9-4746-ac92-09cb04a4e7c6)

Linux desktop:
![image](https://github.com/raysan5/raylib/assets/28019266/08938d42-7513-4dc2-a3f1-c2357b71f53f)

### Code Example

```
#include "raylib.h"

#if defined(PLATFORM_WEB)
#include <emscripten/emscripten.h>
#endif

void game_frame(void *data) {

  BeginDrawing();
  ClearBackground(RED);

  if (IsKeyPressed(KEY_N)) {
    TraceLog(LOG_INFO, "Works on both");
  }
  EndDrawing();

  if (IsKeyPressed(KEY_N)) {
    TraceLog(LOG_INFO, "Does not work in browser");
  }
}

int main(void) {
  InitWindow(200, 200, "issue");

  int data;

#if defined(PLATFORM_WEB)
  emscripten_set_main_loop_arg((void (*)(void *))game_frame, &data, 0, 1);

#else
  SetTargetFPS(60);
  // Main game loop
  while (!WindowShouldClose())
  {
    game_frame(&data);
  }
#endif

  CloseWindow();

  return 0;
}
```

**Comment by kallehed** [comment]
![image](https://github.com/raysan5/raylib/assets/28019266/60e998a6-1c9a-4718-a70f-58735e9afe2c)

The CORE.Input.Keyboard.previousKeyState[key] must have changed to a 1 after EndDrawing.

**Comment by kallehed** [comment]
I have done some investigations now:
On desktop: The previous key state will be 0 for both BEFORE EndDrawing and right AFTER EndDrawing, while the current key state will be 1.
That means that IsKeyPressed() will return 1 for both cases.
BUT, on PLATFORM_WEB, the previous key state will be 0 BEFORE EndDrawing, but 1 AFTER EndDrawing. While the current key state will be 1 for both.

**Comment by kallehed** [comment]
So, for some reason PollInputEvents is not called on Desktop after EndDrawing?

**Comment by kallehed** [comment]
Oh wait, that's not it. I think the problem is just that events are getting updated at different times? On Desktop the keypress is registered immediately after EndDrawing, which allows the TraceLog "does not print in browser" to run, but on the Web, the event is received before the main loop is started, and thus the first TraceLog will print, but then after EndDrawing, the key N will be marked as not "fresh" anymore, and be delegated to previous keys pressed. 

Basically, the glfw callback seems to run before/after the main loop in the Web. But on desktop, I think it's being run some other time.

**Comment by kallehed** [comment]
Yes, this is exactly what is happening. The callback is running at different times for Desktop and the Web.

**Comment by kallehed** [comment]
Ah, this is because, ONLY on desktop: in PollInputEvents glfwPollEvents() is called
On PLATFORM_WEB, glfwPollEvents seems to NOT be called at all. Presumably handled somewhere else then, or automatically by glfw?

**Comment by kallehed** [comment]
![image](https://github.com/raysan5/raylib/assets/28019266/faf1d243-f963-4680-9e68-082b8ab42532)
specifically this part

**Comment by kallehed** [comment]
Ok, this is because of: https://github.com/emscripten-core/emscripten/blob/main/src/library_glfw.js
It is implemented by instantly calling the key callback when a key is pressed. This is not how normal GLFW works.
Therefore, because my main loop is so small and fast, in the majority of times, events will be received outside my main loop. Which is the opposite of the normal behaviour (Events received at EndDrawing()).
So then, the question becomes, how do we fix this exactly?

**Comment by kallehed** [comment]
Either Emscripten fix their version of GLFW (best case)
or, Raylib could somehow introduce buffering code that introduces a layer between the glfw callback and it's own key state. But this feels kind of complex and annoying.
Or, we can provide a warning, that IsKeyPressed should not be called after EndDrawing, as it will have different behaviour depending on when the key callbacks are called.

**Comment by kallehed** [comment]
Yes, exactly. Though, where in the documentation should the note be put?

**Comment by kallehed** [comment]
Yes, though I have not seen that page before, I agree that it is quite suitable for this kind of information. 
Maybe we could write something under the PLATFORM_WEB header like: "On PLATFORM_WEB key-event information is updated at random times, instead of at EndDrawing, therefore we suggest only reading events before EndDrawing (in your main loop) to make behaviour consistent between other platforms."

**Comment by raysan5** [comment]
> Changing that looks like a sizeable change that could break a lot on the way.

Absolutely. Also seems an issue related to `libray_glfw.js` implementation while trying to replicate desktop GLFW.

 > IMHO, the documentation on the wiki (which contains the library design and architectural information) would still be the ideal place.

I think it's the best approach for this issue.

@kallehed @ubkp Feel free to review the Wiki to add this implementation detail.

I'm closing this issue, any further implementation is out of scope.

**Comment by kallehed** [comment]
Ok, I added a warning to the wiki.
